bot/cogs/reddit.py

```diff
@@ -113,103 +120,63 @@ def __init__(self, bot: Bot):
             )
 
         embed.colour = Colour.blurple()
+        return embed
 
-        return await channel.send(
-            content=content,
-            embed=embed
-        )
-
-    async def poll_new_posts(self) -> None:
-        """Periodically search for new subreddit posts."""
-        while True:
-            await asyncio.sleep(RedditConfig.request_delay)
-
-            for subreddit in RedditConfig.subreddits:
-                # Make a HEAD request to the subreddit
-                head_response = await self.bot.http_session.head(
-                    url=f"{self.URL}/{subreddit}/new.rss",
-                    headers=self.HEADERS
-                )
-
-                content_length = head_response.headers["content-length"]
-
-                # If the content is the same size as before, assume there's no new posts.
-                if content_length == self.prev_lengths.get(subreddit, None):
-                    continue
-
-                self.prev_lengths[subreddit] = content_length
-
-                # Now we can actually fetch the new data
-                posts = await self.fetch_posts(f"{subreddit}/new")
-                new_posts = []
+    @loop()
+    async def top_daily_posts_loop(self) -> None:
+        """Post the top 5 posts daily."""
+        # once we upgrade to d.py 1.3 this can be removed and the loop can use the `time=datetime.time()` parameter
+        now = datetime.utcnow()
 
-                # Only show new posts if we've checked before.
-                if subreddit in self.last_ids:
-                    for post in posts:
-                        data = post["data"]
+        midnight_tomorrow = now.replace(day=now.day + 1, hour=0, minute=0, second=0)
+        seconds_until = (midnight_tomorrow - now).total_seconds()
 
-                        # Convert the ID to an integer for easy comparison.
-                        int_id = int(data["id"], 36)
+        await asyncio.sleep(seconds_until)
```

Comment:
The `discord.py` Tasks extensions allows you to pass a single (or a list of) `datetime.time` to the `time` parameter of `loop()` to trigger an iteration at the specified time. That way, we don't have to manually calculate `seconds_until` and include a `asyncio.sleep`. 

I think this should do it:

```py
import datetime
@loop(time=datetime.time.min)
```

The use of `datetime.time.min` guarantees we're aiming for the midnight exactly, although I don't think a few seconds really matter.

Reply:
I absolutely agree, although the `time` parameter is new in version 1.3 - were in v1.2.3, unless we wish to move to v1.3a (I see no problem, personally).

I've added a note in the comments that this should be fixed once we upgrade